test/suites/integration/msp/debt-collection.test.ts

```diff
@@ -145,22 +176,47 @@ describeMspNet("Single MSP collecting debt", ({ before, createMspApi, it, create
       "Expected BucketRootChanged event but received event of different type"
     );
 
+    // Ensure the new root of the bucket matches the one in the event
     strictEqual(bucketRootChangedDataBlob.newRoot.toString(), localBucketRoot.toString());
 
+    // Ensure that the file has been stored in the MSP's forest, in the bucket's trie
     const isFileInForest = await mspApi.rpc.storagehubclient.isFileInForest(
       newBucketEventDataBlob.bucketId.toString(),
       acceptedFileKey
     );
 
     assert(isFileInForest.isTrue, "File is not in forest");
 
+    // Check that the rate of the payment stream between the user and the MSP has been updated
+    // to reflect the new size of the bucket
+    paymentStream = (
+      await userApi.query.paymentStreams.fixedRatePaymentStreams(
+        DUMMY_MSP_ID,
+        userApi.shConsts.NODE_INFOS.user.AddressId
+      )
+    ).unwrap();
+    const bucketOption: Option<H256> = userApi.createType("Option<H256>", bucketId);
+    const firstFileSize = (
+      await mspApi.rpc.storagehubclient.getFileMetadata(bucketOption, acceptedFileKey)
+    )
+      .unwrap()
+      .file_size.toNumber();
+    const unitsInGigaUnit = 1024 * 1024 * 1024;
+    let expectedPaymentStreamRate =
+      (valueProps[0].value_prop.price_per_giga_unit_of_data_per_block.toNumber() * firstFileSize) /
+        unitsInGigaUnit +
+      zeroSizeBucketFixedRate;
```

Comment:
note: it would be nice to have an RPC method for users to be able call and return the cost of adding a file with some size to some bucket. then dApps can expose this information when users are about to add files via some dashboard.